### PR TITLE
Rad/spool 352/update chef index

### PR DESCRIPTION
**Pull request by rmoshier**
Most of this is copying files over from oc_erchef so reporting can work correctly with chef_index. Things of note include: 

**First commit**
- Information transfer from chef_server

**Second commit**
- Removed chef_index batch functionality because oc_reporting doesn't need to submit data
- README update
- Update rebar.config
- Editing tests to appease dialyzer

Equivalent Chef server README update: https://github.com/chef/chef-server/pull/937

### CODE REVIEW

File: src/chef_index.app.src [hunk 1]
```diff
@@ -18,12 +18,11 @@
 {application, chef_index,
  [
   {description, "Chef search index tools"},
-  {vsn, git},
+  {vsn, {cmd,"cat ../../VERSION | awk '{print $0}'"}},
```
Comment (stevendanna) [inline src/chef_index.app.src]: This bit assumes the chef-server's setup re the VERSION file, will that still make sense in the context it is being used?  Maybe we can change this to a static version number if it doesn't.

File: test/chef_index_expand_tests.erl [hunk 2]
```diff
@@ -64,7 +64,12 @@ flatten_lists_test() ->
     ?assertEqual(Expect, iolist_to_binary(Expanded)).
 
 example_test() ->
-    {ok, Bin} = chef_index_test_utils:read_file("sample_node.json"),
+    % {ok, Bin} = chef_index_test_utils:read_file("sample_node.json"),
+    io:fwrite(filename:absname("sample_node.json"), []),
```
Comment (stevendanna) [inline test/chef_index_expand_tests.erl]: This looks like it is potentially debugging output?

File: src/chef_index.erl [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        _ -> %% batch mode not implemented for delete, always use inline if not rabbitmq
+            stats_hero:ctime(ReqId, {chef_solr, delete},
+                             fun() ->
+                                     Doc = chef_index_expand:doc_for_delete(TypeName, Id, DbName),
+                                     chef_index_expand:send_delete(Doc)
+                             end)
+    end.
+
+queue_mode() ->
+    envy:get(chef_index, search_queue_mode, rabbitmq, envy:one_of([rabbitmq, batch, inline])).
```
Comment (stevendanna) [inline src/chef_index.erl]: Do we think there would be any benefit of completely excising the batch mode from this code?

**Review by stevendanna — COMMENTED** [review]
I've left a few comments.  I think given our current expectations of oc_reporting's future development, I think this path makes sense.  Part of me wonders whether we should do this on a different branch, but on the other hand, I think surfacing other projects that might be pointed at master of this deprecated repo without a lock would be a good thing.

File: src/chef_index.app.src (hunk 1 above)
Comment (marcparadise) [inline src/chef_index.app.src]: I think we can leave this as the original `git` value.

File: src/chef_index.erl (hunk 3 above)
Comment (marcparadise) [inline src/chef_index.erl]: For this specific case, I think less is more.   Minimal changes will make it easier to keep in sync with`chef_index` in the chef-server repo, if it becomes necessary to do so.

**Comment by marcparadise** [comment]
:+1:

**Pull request merged**